### PR TITLE
Add different mode for transaction header override

### PR DESCRIPTION
**Pull request by gaetanmaisse**
## Issue

https://gravitee.atlassian.net/browse/APIM-75
https://github.com/gravitee-io/issues/issues/7618

## Description

Response headers were previously handled in the TransactionRequestProcessor. 
They are now processed in the TransactionResponseProcessor with specific configuration keys, see below for details ⬇️ 

To avoid introducing another breaking change it would be interesting to let the users pick the behavior they want when the backend of the API is also setting the `request` or `transaction` header. So, the configuration will be updated to something like this:

```yaml
gravitee:
  handlers:
    request:
      format: uuid
      request:
        header: X-CUSTOM-Flow-ID
        # ⬇️ This is a new attribute
        overrideMode: none 
      transaction:
        header: X-CUSTOM-Flow-ID
        # ⬇️  This is a new attribute
        overrideMode: none
```

Possible values of `overrideMode` are the following:
 - `override`: The header set by the APIM Gateway will be kept and the one provided by the backend discarded 
 - `merge`: Both headers set by the APIM Gateway and the backend will be kept (as headers can be multi-valued)
 - `keep`: The header set by the APIM Gateway will be overridden by the one provided by the backend

The `override` mode will be set by default.

## Additional context

🚀 CI was able to deploy the build of this PR, so you can now try it directly [here](https://apimnightlywebui24386.z6.web.core.windows.net/apim-75-add-backendoverridemode/index.html)
_Notes_: The deployed app is linked to the management API of APIM master. (Same login and password as APIM master)

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-lracixmayq.chromatic.com)

### CODE REVIEW

**Comment by gaetanmaisse** [comment]
Fix on the Jupiter side will come with the next PR (on 3.19.x +)

File: gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/processor/ResponseProcessorChainFactory.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import java.util.Arrays;
 import java.util.List;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
```
Comment (a-cordier) [inline gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/processor/ResponseProcessorChainFactory.java]: I guess this import can be removed ?

**Review by a-cordier — APPROVED** [review]

File: gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/processor/ResponseProcessorChainFactory.java (hunk 1 above)
Comment (gaetanmaisse) [reply gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/processor/ResponseProcessorChainFactory.java]: True

**Comment by a-cordier** [comment]
✅ tested

**Pull request merged**